### ISSUE TITLE
Underline drawn up to current bspwm workspace when fallback and module settings are defined

### ISSUE THREAD

**Issue by xeals** [polybar/polybar#598]
Apologies for the title gore.

If I have a `format-underline` defined in the `[settings]` block then again for the focused workspace in the bspwm module, it draws the underline up to the current workspace instead of for only the focused one. I don't have this issue if I define an underline for at least one other module.

edit: on second glance at it, ~I didn't have anything on the first workspace in the second image. The occupied underline also writes over the first workspace if the empty underline isn't defined, and the second if it's active. Then if I have the empty underline defined, the active workspace only gets an underline if it's the first one.~ there's a bit more going on. I didn't have a very isolated environment, so testing is down the bottom.

Relevant config:

```
[settings]
format-underline = ${colors.fg}
...
[bspwm]
[module/bspwm]
type = internal/bspwm
format = <label-state>

label-focused-foreground = ${colors.fg}
label-focused-background = ${colors.color12}
label-focused-underline = ${colors.fg}
label-focused-padding-right = 1
label-occupied-foreground = ${colors.colorx}
label-occupied-padding-right = 1
; label-occupied-underline = ${colors.bg} ; used in second image
label-empty-foreground = ${colors.color8}
label-empty-padding-right = 1
label-urgent-foreground = ${colors.color1}
label-urgent-padding-right = 1
```

With only the focused underline:

![2017-05-30-111554_234x31_scrot](https://cloud.githubusercontent.com/assets/21125058/26564880/9b7edc66-4529-11e7-919c-71f5dea915c9.png)

Adding the occupied underline:

![2017-05-30-111626_227x26_scrot](https://cloud.githubusercontent.com/assets/21125058/26564881/9b7f83be-4529-11e7-9cb1-b3488f8710f5.png)

addition: Sort of flowchart of states and effects.

| state | result |
| --- | --- |
| only focused | default line to current workspace, current workspace as defined |
| only occupied | default line to first occupied workspace |
| only empty | default line as long as the workspaces are unoccupied (so none if 1 is unoccupied, 2 if 1 and 2 are occupied, etc.) |
| focused + occupied | default line as long as workspaces are empty |
| focused + empty | default line as long as workspaces are occupied |
| occupied + empty | default line if the first workspace is occupied, otherwise nothing |

tl;dr after some revision, it looks like the default line is only applying until the first case where it shouldn't, and ignoring further cases.

**Comment by patrick96** [comment]
I think this issue is similar to #937. Because `label-focused` has the underline set, it disables the underline after the focused workspace. That's why the underline stops at the focused workspace. If you add occupied underline the very first workspace already resets the underline, that's why it looks like it's working correctly.

So we need to also stack the activations of overline and underline.

What was your expected behavior here? That the whole module is underlined with the color from `[settings]` and the different workspace types can use different colors?

**Comment by xeals** [comment]
I honestly don't know what I was expecting the behaviour to be at this point in time. It's been 18 months and I haven't used that configuration in just as long. Probably easiest to close this issue and move on unless someone has the exact same question about it.

**Comment by patrick96** [comment]
I'm gonna leave this open. I'm actually working on a proposal to make the whole formatting system more well-defined and not cause such weird issues.

**Comment by patrick96** [comment]
Closing in favor of #937
